.github/workflows/release-dotnet.yml

```diff
@@ -2,25 +2,62 @@ name: Release .NET SDK
 
 on:
   workflow_dispatch:
-  release:
-    types: [published]
+  # release:
+  #   types: [published]
+  workflow_run:
+    workflows: ["Release Platform Libraries"]
+    branches: [main]
+    types:
+      - completed
 
 jobs:
+  package_nuget:
+    name: Package Nuget
+    runs-on: windows-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v1
+        with:
+          submodules: "true"
+      - name: Download workflow artifact
+        uses: dawidd6/action-download-artifact@v2.14.0
+        with:
+          workflow: "build-libs.yml"
+          path: ./libs
+      - name: Setup MSBuild.exe
+        uses: microsoft/setup-msbuild@v1.0.2
+      - name: Generate Package Version
+        run: |
+          $json = Invoke-WebRequest 'https://api.github.com/repos/trinsic-id/okapi/releases/latest' | ConvertFrom-Json
+          $version = $json.tag_name.Trim("v")
+
+          echo "PACKAGE_VERSION=$version" | Out-File -FilePath $env:GITHUB_ENV -Encoding utf-8 -Append
+        shell: pwsh
+
+      - name: Create Nuget package
+        run: |
+          msbuild -target:restore,pack -property:SkipNativeLibsCopy=true -property:Configuration=Release -property:PackageVersion=$env:PACKAGE_VERSION
+        shell: pwsh
+        working-directory: ./dotnet/Library/Okapi
+
+      - name: Upload nuget artifact
+        uses: actions/upload-artifact@v2
+        with:
+          name: nuget
+          path: |
+            ./dotnet/Library/Okapi/bin/Release/*.nupkg
+            ./dotnet/Library/Okapi/bin/Release/*.snupkg
+
   publish_nuget:
     name: Publish Nuget
-    if: |
-      github.event_name == 'release' &&
-      github.event.action == 'published' &&
-      github.release.draft == false &&
-      github.release.prerelease == false
+    environment:
+      name: nuget.org
     runs-on: windows-latest
     steps:
       - name: Download workflow artifact
-        uses: dawidd6/action-download-artifact@v2.14.0
+        uses: actions/download-artifact@v2
         with:
-          workflow: "build-dotnet.yml"
-          path: ./nuget
-          branch: main
+          name: nuget
```

Comment:
Nice! I'll make a PR that does similar things for the other languages.

Reply:
Let's do few iterations of this process first on the .NET stream before we apply it. I've some ideas I wanna run by you